src/openzeppelin/token/erc721.cairo

```diff
@@ -0,0 +1,358 @@
+            _operator_approvals::read((owner, operator))
+        }
+
+        fn approve(to: ContractAddress, token_id: u256) {
+            let owner = _owner_of(token_id).expect('ERC721: invalid token ID');
+
+            let caller = get_caller_address();
+            assert(
+                owner == caller | is_approved_for_all(owner, caller),
+                'ERC721: unauthorized caller'
+            );
+            _approve(to, token_id)
+        }
+
+        fn set_approval_for_all(operator: ContractAddress, approved: bool) {
+            _set_approval_for_all(get_caller_address(), operator, approved)
+        }
+
+        fn transfer_from(from: ContractAddress, to: ContractAddress, token_id: u256) {
+            assert(
+                _is_approved_or_owner(get_caller_address(), token_id),
+                'ERC721: unauthorized caller'
+            );
+            _transfer(from, to, token_id);
+        }
+
+        fn safe_transfer_from(
+            from: ContractAddress, to: ContractAddress, token_id: u256, data: Array<felt252>
+        ) {
+            assert(
+                _is_approved_or_owner(get_caller_address(), token_id),
+                'ERC721: unauthorized caller'
+            );
+            _safe_transfer(from, to, token_id, data);
+        }
+    }
+
+    #[view]
+    fn supports_interface(interface_id: felt252) -> bool {
+        erc165::ERC165Contract::supports_interface(interface_id)
+    }
+
+    #[view]
+    fn name() -> felt252 {
+        ERC721::name()
+    }
+
+    #[view]
+    fn symbol() -> felt252 {
+        ERC721::symbol()
+    }
+
+    #[view]
+    fn token_uri(token_id: u256) -> felt252 {
+        ERC721::token_uri(token_id)
+    }
+
+    #[view]
+    fn balance_of(account: ContractAddress) -> u256 {
+        ERC721::balance_of(account)
+    }
+
+    #[view]
+    fn owner_of(token_id: u256) -> ContractAddress {
+        ERC721::owner_of(token_id)
+    }
+
+    #[view]
+    fn get_approved(token_id: u256) -> ContractAddress {
+        ERC721::get_approved(token_id)
+    }
+
+    #[view]
+    fn is_approved_for_all(owner: ContractAddress, operator: ContractAddress) -> bool {
+        ERC721::is_approved_for_all(owner, operator)
+    }
+
+    #[external]
+    fn approve(to: ContractAddress, token_id: u256) {
+        ERC721::approve(to, token_id)
+    }
+
+    #[external]
+    fn set_approval_for_all(operator: ContractAddress, approved: bool) {
+        ERC721::set_approval_for_all(operator, approved)
+    }
+
+    #[external]
+    fn transfer_from(from: ContractAddress, to: ContractAddress, token_id: u256) {
+        ERC721::transfer_from(from, to, token_id)
+    }
+
+    #[external]
+    fn safe_transfer_from(
+        from: ContractAddress, to: ContractAddress, token_id: u256, data: Array<felt252>
+    ) {
+        ERC721::safe_transfer_from(from, to, token_id, data)
+    }
+
+    #[internal]
+    fn initializer(name_: felt252, symbol_: felt252) {
```

Comment:
Hmm are the `internal` and `private` macros included to make the contract more readable?

Reply:
yes

Reply:
I like it. I think it'd be a bit weird to only have these macros in erc721 though. @martriay thoughts on updating your PRs?

Reply:
done